yt/frontends/amrvac/fields.py

```diff
@@ -59,64 +64,89 @@ def _get_ekin(data):
                 ekin = ekin + 0.5 * data["amrvac", "m2"] ** 2 / data["amrvac", "rho"]
             if self.ds.dimensionality > 2:
                 ekin = ekin + 0.5 * data["amrvac", "m3"] ** 2 / data["amrvac", "rho"]
+            # override units, everything is normalised
+            ekin.units = self.ds.unit_system['specific_energy']
             return ekin
+
         def _get_emag(data):
             emag = 0.5 * data["amrvac", "b1"]**2
             if self.ds.dimensionality > 1:
                 emag = emag + 0.5 * data["amrvac", "b2"]**2
             if self.ds.dimensionality > 2:
                 emag = emag + 0.5 * data["amrvac", "b3"]**2
+            emag.units = self.ds.unit_system['specific_energy']
             return emag
+
         def _pressure(field, data):
             # (M)HD datasets
             if ("amrvac", "e") in self.field_list:
                 # MHD dataset
                 if ("amrvac", "b1") in self.field_list:
-                    return (self.ds.gamma - 1) * (data["amrvac", "e"] - _get_ekin(data) - _get_emag(data))
+                    pres = (self.ds.gamma - 1) * (data["amrvac", "e"] - _get_ekin(data) - _get_emag(data))
                 # Non-MHD dataset
                 else:
-                    return (self.ds.gamma - 1) * (data["amrvac", "e"] - _get_ekin(data))
+                    pres = (self.ds.gamma - 1) * (data["amrvac", "e"] - _get_ekin(data))
             else:
+                # TODO: define unit_pressure in case there is no energy equation (right now it's code_pressure above)
                 # Non (M)HD datasets, in this case an EoS is used for the pressure
-                return adiab_cte * data["amrvac", "rho"] ** self.ds.parameters.gamma
+                pres = adiab_cte * data["amrvac", "rho"] ** self.ds.parameters.gamma
+            pres.units = self.ds.unit_system['pressure']
+            return pres
+
         def _temperature(field, data):
-            return data["amrvac", "pressure"] / data["amrvac", "rho"]
+            temp = data["amrvac", "pressure"] / data["amrvac", "rho"]
+            temp.units = self.ds.unit_system['temperature']
+            return temp
+
         def _velocity1(field, data):
-            return data["amrvac", "m1"] / data["amrvac", "rho"]
+            v1 = data["amrvac", "m1"] / data["amrvac", "rho"]
+            v1.units = self.ds.unit_system['velocity']
+            return v1
+
         def _velocity2(field, data):
-            return data["amrvac", "m2"] / data["amrvac", "rho"]
+            v2 = data["amrvac", "m2"] / data["amrvac", "rho"]
+            v2.units = self.ds.unit_system['velocity']
+            return v2
+
         def _velocity3(field, data):
-            return data["amrvac", "m3"] / data["amrvac", "rho"]
+            v3 = data["amrvac", "m3"] / data["amrvac", "rho"]
+            v3.units = self.ds.unit_system['velocity']
+            return v3
+
```

Comment:
units should not have to be imposed manually. The mistake here is that native fields haven't been set to their correct respective dimensions